dbt/adapters/duckdb/environments/local.py

```diff
@@ -100,20 +102,29 @@ def load_source(self, plugin_name: str, source_config: utils.SourceConfig):
                 params.append(source_config.database)
             if cursor.execute(q, params).fetchone()[0]:
                 if save_mode == "error_if_exists":
-                    raise Exception(f"Source {source_config.table_name()} already exists!")
+                    raise Exception(
+                        f"Source {source_config.table_name()} already exists!"
+                    )
                 else:
                     # Nothing to do (we ignore the existing table)
                     return
         df = plugin.load(source_config)
         assert df is not None
-        materialization = source_config.meta.get("materialization", "table")
-        cursor.execute(
-            f"CREATE OR REPLACE {materialization} {source_config.table_name()} AS SELECT * FROM df"
-        )
+
+        if plugin_name == "delta": # potentially should all plugins use configure_cursor
```

Comment:
this block seems redundant now, no? Aren't all cursors getting configured by plugins?

Reply:
Hm i am not sure if i understand what you think. There are multiple things why i wanted to move that code out of general flow:
* The delta plugin always want a VIEW and not table as materialization
* I cant create a view with such a general df because i have to register views under different names -> therefore i moved the whole view creation block inside

Those are the reasons why i couldn't put my code there but just to coll configure_cursor in my opinion the plugins which wants and are okey with general flow to return df and load it with that materialization should/can use else branch and others which need more configuration go into `plugin in ("delta",....)` branch and configure each cursor. This also leaves the room for backportability that i dont have to change something by other plugins till it is not needed.     

Just a note i am not sure but i have feeling that if we for example in excel plugin allow materialization to be view that it will have the same problem as i had that df is not registered so this can maybe be a potential bug

Reply:
Ah, I think I see the confusion-- so you cannot use a plugin (like delta) as the write-path for a regular dbt python model (which, as you noted, always requires a table.) You can only use a plugin with the dbt-duckdb specific `external` materialization, which ALWAYS creates a view in DuckDB on top of the data files that it writes. So for the write-side of delta, you want to implement the `store` method on your Plugin class, which gets called here:

https://github.com/duckdb/dbt-duckdb/blob/master/dbt/adapters/duckdb/impl.py#L80

...and it should be responsible for loading the parquet files that were written into a a delta table.

Reply:
I am not so sure that if i understood. 
There are two topics one is your question
* Aren't all cursors getting configured by plugins? -> and after rereading if i understand it correctly you want to say that this block with configure_cursor is redudant because it will be configured by new cursor anyway. -> This is true you were right and i deleted that code block

and another one is:
* In your last comment above you are speaking about writing to the delta table? -> i would like that we do first just the reading in this PR and i have to do first a concept how to write to delta tables because i think it will not be trivial and there is also some knowlage missing on my side how actually the adapter and sql execution is interchanging and compiled. Therefore i would move that in separate PR if thats okey for you.

Reply:
Ah that is totally okay with me, thanks!